### ISSUE TITLE
Problems with multiple disjoint selections

### ISSUE THREAD

**Issue by nikhilweee** [Simonwep/selection#126]
**Is your feature request related to a problem? Please describe.**

I'm using selection JS in a different manner, so I'm not sure whether my problem is worthy of a feature request. 

**Additional context**

I'm using selectionJS on a large canvas of more than 1000 items, and I want to be able to make multiple disjoint selections. According to the current behaviour, all previous selections are lost when someone makes a new selection or clicks on an empty area, so I wrote the following to bypass that behaviour.

```js
selection.on('start', ({ event, store }) => {
    let target = event.target.parentNode;
    if (target.classList.contains('selected')) {
        target.classList.remove('selected');
        store.changed.removed.push(target);
    }
}).on('move', ({ store }) => {
    for (const el of store.changed.added) {
        el.classList.add('selected');
    }
    for (const el of store.changed.removed) {
        el.classList.remove('selected');
    }
}).on('stop', ({ store }) => {
    selection.keepSelection();
    store.stored = store.stored.filter((item, index, self) =>
        index === self.findIndex((t) => (
            t.innerText === item.innerText
        ))
    )
});
```
Now, I want to be able to use SHIFT key to make multiple disjoint selections. Essentially, I want the following behaviour.
1. I click on one item 5. 
2. I hold the SHIFT key, select item 10. 
At this point, I want all items 5, 6, 7, 8, 9, 10 to be selected. This works fine. But since I enabled multiple selections, I want the following to work as well.
3. I select item 15. My previous selection stays intact. (This works fine)
4. I hold the SHIFT key, select item 20.
At this point, I only want items 15, 16, 17, 18, 19, 20 to be selected. But right now, all items between 5 to 20 are being selected.

**Describe the solution you'd like**

The problem arises because the `reference` is being chosen as the first element ever to be selected. If this is changed to the most recently selected element, my problem is solved.
https://github.com/Simonwep/selection/blob/7d36bbdb504e2e752794d7375207b9eaf821fe31/src/index.ts#L214
Essentially, changing the line above to the line below works well.
```js
    const reference = stored[0]; 
```

**Describe alternatives you've considered**

I did not consider any alternate solutions.

Although I'm not asking to implement multiple disjoint selections, that might be a cool feature to have.

---
## TL;DR
Please consider changing the following:
https://github.com/Simonwep/selection/blob/7d36bbdb504e2e752794d7375207b9eaf821fe31/src/index.ts#L214
To the following:
```js
    const reference = stored[0]; 
```
Thanks!

**Comment by Simonwep** [comment]
Actutally, I just checked - this is the behaviour of windows / linux... Guess I'm gonna change it then :D
I'll release a new beta version later this week if this is okay with you, in case anything else comes up and / or into my mind.
